pilot/pkg/xds/delta.go

```diff
@@ -245,7 +245,7 @@ func (conn *Connection) sendDelta(res *discovery.DeltaDiscoveryResponse) error {
 	}
 	err := istiogrpc.Send(conn.deltaStream.Context(), sendHandler)
 	if err == nil {
-		if res.Nonce != "" && !strings.HasPrefix(res.TypeUrl, v3.DebugType) {
```

Comment:
Why remove the non empty response nonce check?

Reply:
Because it can never be empty, we set nonce in `pushDeltaXds` before calling `sendDelta`